Sources/SwiftDocC/Utility/FileManagerProtocol+FilesSequence.swift

```diff
@@ -0,0 +1,66 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+extension FileManagerProtocol {
+    /// Returns a sequence of all the files in the directory structure from the starting point.
+    package func recursiveFiles(startingPoint: URL) -> some Sequence<URL> {
+        IteratorSequence(FilesIterator(fileManager: self, startingPoint: startingPoint))
+    }
+}
+
+/// An iterator that traverses the directory structure and returns the files in breadth-first order.
+private struct FilesIterator<FileManager: FileManagerProtocol>: IteratorProtocol {
+    /// The file manager that the iterator uses to traverse the directory structure.
+    var fileManager: FileManager
+    
+    private var foundFiles: [URL]
+    private var foundDirectories: [URL]
+    
+    init(fileManager: FileManager, startingPoint: URL) {
+        self.fileManager = fileManager
+        
+        // Check if the starting point is a file or a directory.
+        if fileManager.directoryExists(atPath: startingPoint.path) {
+            foundFiles       = []
+            foundDirectories = [startingPoint]
+        } else {
+            foundFiles       = [startingPoint]
+            foundDirectories = []
+        }
+    }
+    
+    mutating func next() -> URL? {
+        // If the iterator has already found some files, return those first
+        if !foundFiles.isEmpty {
+            return foundFiles.removeFirst()
+        }
+        
+        // Otherwise, check the next found directory and add its contents
+        guard !foundDirectories.isEmpty else {
+            // Traversed the entire directory structure
+            return nil
+        }
+        
+        let directory = foundDirectories.removeFirst()
+        guard let (newFiles, newDirectories) = try? fileManager.contentsOfDirectory(at: directory, options: .skipsHiddenFiles) else {
```

Comment:
Maybe `options` should be an option passed into the initializer, with `skipsHiddenFiles` as the default value.

Reply:
Added in https://github.com/swiftlang/swift-docc/pull/1108/commits/e1f229cf03d00d0e99d1da7d2dca19b0c22c07ef